### ISSUE TITLE
Gateway-API: Istio sidecar container not found in Pod(s)

### ISSUE THREAD

**Issue by obervinov** [kiali/kiali#7860]
### Describe the bug

When using gateway-api with `gatewayClass: istio`, kiali reports a warning that sidecar is not found on the gateway pod
![Снимок экрана 2024-10-24 в 16 59 57](https://github.com/user-attachments/assets/d76baf55-0afe-4a43-b786-97fc0f0a78c9)
This is strange because the default pod gateway includes an istio-proxy container.
I suspect it's the `sidecar.istio.io/inject: ‘false’` label that the istio mutation webhook adds automatically when creating a pod for a gateway.

### Expected Behavior

I think the gateway-api case should not have such a warning, it is misleading

### What are the steps to reproduce this bug?

1. Create a gateway in the cluster
```yaml
---
apiVersion: gateway.networking.k8s.io/v1
kind: Gateway
metadata:
  annotations:
    gateway.istio.io/name-override: igw
    networking.istio.io/service-type: ClusterIP
  labels:
    app.kubernetes.io/name: gateway-api
    app.kubernetes.io/version: 1.1.0
  name: igw
spec:
  gatewayClassName: istio
  listeners:
  - allowedRoutes:
      namespaces:
        from: Selector
        selector:
          matchExpressions:
          - key: kubernetes.io/metadata.name
            operator: In
            values:
            - default
    hostname: igw.default
    name: http
    port: 80
    protocol: HTTP
```
2. After istio using mutation webhook automatically adds the service and pod for this gateway, look at the automatically added label
```yaml
---
apiVersion: v1
kind: Pod
metadata:
  annotations:
    istio.io/rev: default
    networking.istio.io/service-type: ClusterIP
    prometheus.io/path: /stats/prometheus
    prometheus.io/port: "15020"
    prometheus.io/scrape: "true"
  labels:
    app.kubernetes.io/name: gateway-api
    app.kubernetes.io/version: 1.1.0
    gateway.networking.k8s.io/gateway-name: igw
    istio.io/gateway-name: igw
    service.istio.io/canonical-name: igw
    service.istio.io/canonical-revision: latest
    sidecar.istio.io/inject: "false"
...
```
3. In the web ui Kiali we see that the workload is labelled as ‘no sidecar’.

### Environment
*Learn about how to determine versions [here](https://kiali.io/docs/faq/general/#how-do-i-determine-what-version-i-am-running).*

- **Kiali version:** v1.87.0
- **Istio version:** 1.22.3
- **Kubernetes impl:** AWS EKS
- **Kubernetes version:** v1.29
- **Other notable environmental factors:** used kubernets gateway-api v1.1.0

**Comment by hhovsepy** [comment]
Hi @obervinov , thank you for logging the issue, seems it is a duplicate of https://github.com/kiali/kiali/issues/7720
Can you try please with Kiali v1.89.7, it should be fixed already.

**Comment by obervinov** [comment]
@hhovsepy thanks for the reply.
Yes, it is indeed the same case, I will need some time to test it on a newer version of kiali.

**Comment by hhovsepy** [comment]
Ok @obervinov ,then closing this issue as fixed in Kiali v1.89.7
If the issue still presents, please reopen.
